Assets/LeapMotion/Modules/GraphicRenderer/Scripts/LeapGraphicRenderer.cs

```diff
@@ -217,6 +222,48 @@ public partial class LeapGraphicRenderer : MonoBehaviour, ISerializationCallback
       }
     }
 
+    private void validateSpaceComponent() {
+      var origSpace = _space;
+      
+      var spaces = Pool<List<LeapSpace>>.Spawn();
+      spaces.Clear();
+      try {
+        GetComponents<LeapSpace>(spaces);
+        _space = spaces.Query().FirstOrDefault(s => s.enabled);
+      }
+      finally {
+        spaces.Clear();
+        Pool<List<LeapSpace>>.Recycle(spaces);
+      }
+
+      if (Application.isPlaying
+          && (origSpace != _space || (_space == null && !_lastSpaceWasNull))) {
```

Comment:
Hmmm why iss the _lastSpaceWasNull check needed at all?  Won't the originSpace check cover all possibilities?

Reply:
No, because if the space is deleted, our reference to it will magically become null out from under our feet -- including the "cached" reference. :(